notebooks/models/text-to-text/T5/master_processing.ipynb

```diff
@@ -0,0 +1,530 @@
+      " 6   Topic                  1852 non-null   object\n",
+      " 7   Marked for processing  3482 non-null   object\n",
+      " 8   Question 1             2182 non-null   object\n",
+      " 9   Question 2             420 non-null    object\n",
+      " 10  Question 3             61 non-null     object\n",
+      "dtypes: int64(2), object(9)\n",
+      "memory usage: 327.0+ KB\n"
+     ]
+    }
+   ],
+   "source": [
+    "merged_df.info()"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-08-03T12:29:30.708249Z",
+     "start_time": "2023-08-03T12:29:30.623936Z"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "outputs": [],
+   "source": [
+    "merged_df.dropna(subset='Question 1',inplace=True)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-08-03T12:29:30.708402Z",
+     "start_time": "2023-08-03T12:29:30.645128Z"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/var/folders/7j/mf5p4rjs4pv7s2ctxyh2kzyr0000gn/T/ipykernel_40225/269196921.py:3: FutureWarning: The default value of regex will change from True to False in a future version.\n",
+      "  dff1[text_field] = dff1[text_field].str.replace(r\"http\\S+\", \"\")\n",
+      "/var/folders/7j/mf5p4rjs4pv7s2ctxyh2kzyr0000gn/T/ipykernel_40225/269196921.py:5: FutureWarning: The default value of regex will change from True to False in a future version.\n",
+      "  dff1[text_field] = dff1[text_field].str.replace(r\"@\\S+\", \"\")\n",
+      "/var/folders/7j/mf5p4rjs4pv7s2ctxyh2kzyr0000gn/T/ipykernel_40225/269196921.py:6: FutureWarning: The default value of regex will change from True to False in a future version.\n",
+      "  dff1[text_field] = dff1[text_field].str.replace(r\"[^A-Za-z0-9(),!?@\\'\\`\\\"\\_\\n]\", \" \")\n"
+     ]
+    }
+   ],
+   "source": [
+    "#cleaning text\n",
+    "def standardize_text(dff1, text_field):\n",
+    "    dff1[text_field] = dff1[text_field].str.replace(r\"http\\S+\", \"\")\n",
+    "    dff1[text_field] = dff1[text_field].str.replace(r\"http\", \"\")\n",
+    "    dff1[text_field] = dff1[text_field].str.replace(r\"@\\S+\", \"\")\n",
+    "    dff1[text_field] = dff1[text_field].str.replace(r\"[^A-Za-z0-9(),!?@\\'\\`\\\"\\_\\n]\", \" \")\n",
+    "    dff1[text_field] = dff1[text_field].str.replace(r\"@\", \"at\")\n",
+    "    dff1[text_field] = dff1[text_field].str.replace('\\n','', regex=True)\n",
+    "    dff1[text_field] = dff1[text_field].str.lower()\n",
+    "    return dff1[text_field]\n",
+    "\n",
+    "merged_df['context'] = standardize_text(merged_df, \"context\")\n",
+    "merged_df['Question 1'] = merged_df['Question 1'].str.lower()\n",
+    "merged_df['Question 2'] = merged_df['Question 2'].str.lower()\n",
+    "merged_df['Question 3'] = merged_df['Question 3'].str.lower()"
```

Comment:
Isn't T5 case sensitive?

Reply:
Yes, it is. Do you reckon I should not lowercase all text? I thought it wouldn't make a difference and it is usually a standard practice in nlp. I can of course not do it if it seems beneficial to you.

Reply:
It was practice because in the past the models have not distinguished but for new and case sensitive models, of course, it is important. I think you have to run the fine-tuning again with the corrected training data.

Reply:
alright